mitxpro/utils.py

```diff
@@ -167,6 +167,28 @@ def partition(items, predicate=bool):
     return ((item for pred, item in a if not pred), (item for pred, item in b if pred))
 
 
+def unique(iterable):
+    """
+    Returns a generator containing all unique items in an iterable
+
+    Returns:
+         generator: Unique items in the given iterable
+    """
+    seen = set()
+    return (x for x in iterable if x not in seen and not seen.add(x))
```

Comment:
`seen.add(...)` won't ever return anything so I'm not sure what this test does here

Reply:
`seen.add()` is what adds the item to the set in order to prevent duplicates, and since it always returns `None`, `not seen.add(x)` will always evaluate to `True`, which means the item will be included in the generator